packages/streamr-logger/include/streamr-logger/Logger.hpp

```diff
@@ -1,18 +1,152 @@
-#ifndef STREAMER_UTILS_LOGGER_LOGGER_HPP
-#define STREAMER_UTILS_LOGGER_LOGGER_HPP
+#ifndef STREAMER_UTILS_LOGGER_LOGGER_H
+#define STREAMER_UTILS_LOGGER_LOGGER_H
 
-#include <iostream>
+#include <folly/logging/LogConfig.h>
+#include <folly/logging/LogLevel.h>
+#include <folly/logging/LogMessage.h>
+#include <folly/logging/LogWriter.h>
+#include <folly/logging/LoggerDB.h>
+#include <folly/logging/xlog.h>
+#include "StreamrWriterFactory.hpp"
+#include "StreamrHandlerFactory.hpp"
+
+namespace streamr {
+namespace logger {
+
+namespace {
+
+static const auto envLogLevelName = "LOG_LEVEL";
```

Comment:
Maybe could use constexpr std::string_view STRING_CONSTANT = "String"; for string constants instead, see discussion in https://www.reddit.com/r/cpp/comments/cw35kk/best_practices_for_efficient_string_constants/